translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -9468,32 +9468,27 @@ Value *SPIRVToLLVM::transTrinaryMinMaxExtInst(SPIRVExtInst *ExtInst,
   switch (ExtInst->getExtOp()) {
 
   case FMin3AMD: {
-    // Minimum of three FP values. Undefined result if any NaNs.
-    FastMathFlags FMF;
+    // Middle of three FP values. Undefined result if any NaNs.
+    FastMathFlags FMF = getBuilder()->getFastMathFlags();
     FMF.setNoNaNs();
     getBuilder()->setFastMathFlags(FMF);
```

Comment:
Doesn't this leave the nonans flag set in the builder permanently?

Reply:
I think no, cos Builder gets its FMF set for every instruction.